### PR TITLE
Cleanup foreman el8

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
[test rpm]

File: comps/comps-foreman-rhel7.xml [hunk 1]
```diff
@@ -370,7 +370,7 @@
       <packagereq type="default">puppet-agent-puppet-strings</packagereq>
       <packagereq type="default">puppet-agent-rgen</packagereq>
       <packagereq type="default">puppet-agent-yard</packagereq>
-      <packagereq type="default">python2-keycloak-httpd-client-install</packagereq>
+      <packagereq type="default">python3-keycloak-httpd-client-install</packagereq>
```
Comment (ekohl) [inline comps/comps-foreman-rhel7.xml:373]: Shouldn't EL7 remain on Python 2? At least the spec suggests so

Comment (ehelms) [reply comps/comps-foreman-rhel7.xml]: Changed.

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 2]
```diff
@@ -4,16 +4,21 @@
...
+Release:        2%{?dist}
 Summary:        %{summary}
 
 License:        GPLv3
 URL:            https://github.com/latchset/keycloak-httpd-client-install
 Source0:        https://files.pythonhosted.org/packages/source/k/%{pypi_name}/%{pypi_name}-%{version}.tar.gz
 BuildArch:      noarch
- 
+
+%if 0%{?rhel} == 7
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:15]: Usually, we don't do "if rhel", but have "with_pythonX" macros. see for example https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/client/python-apypie/python-apypie.spec

I wouldn't block on that, but strongly prefer ;)

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 3]
```diff
@@ -2,18 +2,29 @@
 %global pypi_name keycloak-httpd-client-install
 %global summary Tools to configure Apache HTTPD as Keycloak client
 
+%if 0%{?rhel} == 7
+%global with_python2 1
+%else
+%global with_python3 1
+%endif
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:9]:
```suggestion
%if 0%{?rhel} == 7
%bcond_without python2
%bcond_with python3
%else
%bcond_with python2
%bcond_without python3
%endif
```

`bcond_with` means "it defaults to without", and `bcond_without` "it defaults to with", confusing, but that are the rules

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 4]
```diff
@@ -2,18 +2,29 @@
...
 BuildArch:      noarch
- 
+
+%if 0%{?rhel} == 7
 BuildRequires:  python2-devel
 BuildRequires:  python2-setuptools
+%else
+BuildRequires:  python3-devel
+BuildRequires:  python3-setuptools
+%endif
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:27]:
```suggestion
%if %{with python2}
BuildRequires:  python2-devel
BuildRequires:  python2-setuptools
%endif

%if %{with python3}
BuildRequires:  python3-devel
BuildRequires:  python3-setuptools
%endif
```

so technically you can build both in parallel

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 5]
```diff
@@ -25,12 +36,13 @@ libraries and tools which can automate and simplify configuring an
 Apache HTTPD authentication module and registering as a client of a
 Keycloak IdP.
 
+%if 0%{?with_python2}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:39]:
```suggestion
%if %{with python2}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 6]
```diff
@@ -43,17 +55,52 @@ Requires:       %{_bindir}/keycloak-httpd-client-install
 Keycloak is an authentication server. This package contains libraries and
 programs which can invoke the Keycloak REST API and configure clients
 of a Keycloak server.
+%endif
+
+%if 0%{?with_python3}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:60]:
```suggestion
%if %{with python3}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 7]
```diff
@@ -43,17 +55,52 @@ Requires:       %{_bindir}/keycloak-httpd-client-install
...
+of a Keycloak server.
+%endif
 
 %prep
 %autosetup -n %{pypi_name}-%{version}
 # Remove bundled egg-info
 rm -rf %{pypi_name}.egg-info
 
 %build
+%if 0%{?with_python2}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:87]:
```suggestion
%if %{with python2}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 8]
```diff
@@ -43,17 +55,52 @@ Requires:       %{_bindir}/keycloak-httpd-client-install
...
 %py2_build
+%endif
+
+%if 0%{?with_python3}
+%py3_build
+%endif
 
 %install
+%if 0%{?with_python2}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:96]:
```suggestion
%if %{with python2}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 9]
```diff
@@ -43,17 +55,52 @@ Requires:       %{_bindir}/keycloak-httpd-client-install
...
 %autosetup -n %{pypi_name}-%{version}
 # Remove bundled egg-info
 rm -rf %{pypi_name}.egg-info
 
 %build
+%if 0%{?with_python2}
 %py2_build
+%endif
+
+%if 0%{?with_python3}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:91]:
```suggestion
%if %{with python3}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 10]
```diff
@@ -43,17 +55,52 @@ Requires:       %{_bindir}/keycloak-httpd-client-install
...
+%py3_build
+%endif
 
 %install
+%if 0%{?with_python2}
 %py2_install
+%endif
+
+%if 0%{?with_python3}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:100]:
```suggestion
%if %{with python3}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 11]
```diff
@@ -62,12 +109,25 @@ install -c -m 644 doc/keycloak-httpd-client-install.8 %{buildroot}/%{_mandir}/ma
 %doc README.md doc/ChangeLog
 %{_datadir}/%{pypi_name}/
 
+%if 0%{?with_python2}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:112]:
```suggestion
%if %{with python2}
```

File: packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec [hunk 12]
```diff
@@ -62,12 +109,25 @@ install -c -m 644 doc/keycloak-httpd-client-install.8 %{buildroot}/%{_mandir}/ma
...
 %files -n python2-%{pypi_name}
 %{python2_sitelib}/*
 %{_bindir}/keycloak-httpd-client-install
 %{_bindir}/keycloak-rest
 %{_mandir}/man8/*
+%endif
+
+%if 0%{?with_python3}
```
Comment (evgeni) [inline packages/foreman/keycloak-httpd-client-install/keycloak-httpd-client-install.spec:120]:
```suggestion
%if %{with python3}
```

**Comment by evgeni** [comment]
that all said, I am fine with the current python approach, so if you need n ack for that, here it is: ACK.

**Comment by ehelms** [comment]
[test rpm]

**Pull request merged**